ops/testing.py

```diff
@@ -1539,11 +1539,16 @@ def add_layer(
                 if service.override not in ('merge', 'replace'):
                     raise RuntimeError('500 Internal Server Error: layer "{}" has invalid '
                                        '"override" value on service "{}"'.format(label, name))
-                if service.override != 'replace':
-                    raise RuntimeError(
-                        'override: "{}" unsupported for layer "{}" service "{}"'.format(
-                            service.override, label, name))
-                layer.services[name] = service
+                elif service.override == 'replace':
+                    layer.services[name] = service
+                elif service.override == 'merge':
+                    if combine and name in layer.services:
+                        s = layer.services[name]
+                        s.merge(service)
+                        layer.services[name] = s
```

Comment:
This is still the same object, so you don't need to reassign it to `layer.services[name]`. The merge has already updated the object in place.